src/toolkit/simulate.ts

```diff
@@ -128,8 +129,20 @@ export function keyPress(
   fireEvent.keyPress(toElement(node), makeKeyEvent(key, props));
 }
 export function insertText(text: string) {
-  // TODO: can this be done via fireEvent?
-  document.execCommand("insertText", false, text);
+  if (
+    document.activeElement.tagName == "textarea" ||
+    document.activeElement.tagName == "input"
+  ) {
+    userEvent.type(document.activeElement, text);
+  } else {
+    if (document.execCommand) {
```

Comment:
It looks like you're defaulting to execCommand here, instead of trying FireEvent first?

Reply:
yeah, the `execCommand` should really just go away in this PR, since now the tests are only running with jsDom. I was just doing both while checking parity across both karma and jest. will update.